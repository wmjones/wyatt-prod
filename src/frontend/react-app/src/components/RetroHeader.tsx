import React from 'react';

interface RetroHeaderProps {
  title: string;
  subtitle?: string;
}

const RetroHeader: React.FC<RetroHeaderProps> = ({ title, subtitle }) => {
  return (
    <div className="text-center mb-12 relative">
      <h1 className="pixel-text text-4xl md:text-6xl font-bold text-retro-purple mb-4 text-shadow animate-bounce-slow">
        {title}
      </h1>
      {subtitle && (
        <p className="retro-text text-2xl md:text-3xl text-retro-teal">
          {subtitle}
          <span className="inline-block ml-1 animate-blink">_</span>
        </p>
      )}
      <div
        data-testid="gradient-divider"
        className="max-w-sm mx-auto my-6 h-2 bg-gradient-to-r from-retro-pink via-retro-purple to-retro-teal">
      </div>
    </div>
  );
};

export default RetroHeader;
