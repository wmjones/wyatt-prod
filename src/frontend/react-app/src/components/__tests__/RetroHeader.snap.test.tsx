import React from 'react';
import renderer from 'react-test-renderer';
import RetroHeader from '../RetroHeader';

describe('RetroHeader Snapshots', () => {
  it('renders correctly with title only', () => {
    const tree = renderer
      .create(<RetroHeader title="Test Title" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with title and subtitle', () => {
    const tree = renderer
      .create(<RetroHeader title="Test Title" subtitle="Test Subtitle" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
