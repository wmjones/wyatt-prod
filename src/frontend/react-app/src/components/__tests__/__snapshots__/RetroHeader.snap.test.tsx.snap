// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetroHeader Snapshots renders correctly with title and subtitle 1`] = `
<div
  className="text-center mb-12 relative"
>
  <h1
    className="pixel-text text-4xl md:text-6xl font-bold text-retro-purple mb-4 text-shadow animate-bounce-slow"
  >
    Test Title
  </h1>
  <p
    className="retro-text text-2xl md:text-3xl text-retro-teal"
  >
    Test Subtitle
    <span
      className="inline-block ml-1 animate-blink"
    >
      _
    </span>
  </p>
  <div
    className="max-w-sm mx-auto my-6 h-2 bg-gradient-to-r from-retro-pink via-retro-purple to-retro-teal"
    data-testid="gradient-divider"
  />
</div>
`;

exports[`RetroHeader Snapshots renders correctly with title only 1`] = `
<div
  className="text-center mb-12 relative"
>
  <h1
    className="pixel-text text-4xl md:text-6xl font-bold text-retro-purple mb-4 text-shadow animate-bounce-slow"
  >
    Test Title
  </h1>
  <div
    className="max-w-sm mx-auto my-6 h-2 bg-gradient-to-r from-retro-pink via-retro-purple to-retro-teal"
    data-testid="gradient-divider"
  />
</div>
`;
